src/app/data_file_manager/data_file_manager.cpp

```diff
@@ -0,0 +1,171 @@
+    FileManager file_manager(path, file_name, FileManager::FileMode::OutputAppend);
+    auto& output_file = file_manager.getFileRef();
+    clinic_data::Workers::Doctors doctor_data;
+    DataFileManager::promptDoctor(doctor_data.add_doctors(), new_doctor);
+    doctor_data.SerializeToOstream(&output_file);
+
+    google::protobuf::ShutdownProtobufLibrary();
+}
+
+void DataFileManager::addReceptionistToDatabase(const std::filesystem::path& path, const std::string& file_name,
+                                                const std::shared_ptr<Receptionist>& new_receptionist)
+{
+    FileManager file_manager(path, file_name, FileManager::FileMode::OutputAppend);
+    auto& output_file = file_manager.getFileRef();
+    clinic_data::Workers::Receptionists receptionist_data;
+    DataFileManager::promptReceptionist(receptionist_data.add_receptionists(), new_receptionist);
+    receptionist_data.SerializeToOstream(&output_file);
+
+    google::protobuf::ShutdownProtobufLibrary();
+}
+
+void DataFileManager::removeDoctorFromFile(const std::filesystem::path& path, const std::string& file_name,
+                                           const std::shared_ptr<Doctor>& doctor)
+{
+
+    clinic_data::Workers::Doctors doctor_data;
+
+    {
+        FileManager file_manager(path, file_name, FileManager::FileMode::Input);
+        auto& input_file = file_manager.getFileRef();
+        if (!doctor_data.ParseFromIstream(&input_file))
+        {
+            throw std::invalid_argument("Error while parsing doctor data");
+        }
+        for (int i = 0; i < doctor_data.doctors_size(); i++)
+        {
+            if (doctor_data.doctors(i).pesel() == doctor->getPesel())
+            {
+                doctor_data.mutable_doctors()->DeleteSubrange(i, 1);
+                break;
+            }
+        }
+    }
+    FileManager file_manager(path, file_name, FileManager::FileMode::OutputTruncate);
+    auto& output_file = file_manager.getFileRef();
+    doctor_data.SerializeToOstream(&output_file);
+
+    google::protobuf::ShutdownProtobufLibrary();
+}
+
+void DataFileManager::removeReceptionistFromFile(const std::filesystem::path& path, const std::string& file_name,
+                                                 const std::shared_ptr<Receptionist>& receptionist)
+{
+    clinic_data::Workers::Receptionists receptionist_data;
+
+    {
+        FileManager file_manager(path, file_name, FileManager::FileMode::Input);
+        auto& input_file = file_manager.getFileRef();
+        if (!receptionist_data.ParseFromIstream(&input_file))
+        {
+            throw std::invalid_argument("Error while parsing receptionist data");
+        }
+        for (int i = 0; i < receptionist_data.receptionists_size(); i++)
+        {
+            if (receptionist_data.receptionists(i).pesel() == receptionist->getPesel())
+            {
+                receptionist_data.mutable_receptionists()->DeleteSubrange(i, 1);
+                break;
+            }
+        }
+    }
```

Comment:
clever with the braces {}, i like this idea, wonder if it's considered in cpp guidlines at all :D

Reply:
I did same as here:
https://protobuf.dev/getting-started/cpptutorial/

Its safer for streams. We can consider blocking streams for safety in the future.